src/pages/DelegationsPage/DelegationsAvailabilityTab/DelegationAvailabilityDrawer.tsx

```diff
@@ -9,34 +9,69 @@ type DelegationAvailabilityDrawerProps = {
   isOpen: boolean
   onClose: VoidFunction
   isAvailableProducerDelegations: boolean
+  isAvailableConsumerDelegations: boolean
 }
 
 export const DelegationAvailabilityDrawer: React.FC<DelegationAvailabilityDrawerProps> = ({
   isOpen,
   onClose,
   isAvailableProducerDelegations,
+  isAvailableConsumerDelegations,
 }) => {
   const { t } = useTranslation('party', { keyPrefix: 'delegations.availabilityTab' })
   const { t: tCommon } = useTranslation('shared-components')
   const { mutate: assignProducerDelegationAvailabilty } =
     TenantMutations.useAssignTenantDelegatedProducerFeature()
   const { mutate: deleteTenantDelegatedProducerFeature } =
     TenantMutations.useDeleteTenantDelegatedProducerFeature()
+  const { mutate: assignConsumerDelegationAvailabilty } =
+    TenantMutations.useAssignTenantDelegatedConsumerFeature()
+  const { mutate: deleteTenantDelegatedConsumerFeature } =
+    TenantMutations.useDeleteTenantDelegatedConsumerFeature()
 
   const [checkedProducerDelegations, setCheckedProducerDelegations] = React.useState(
     isAvailableProducerDelegations
   )
 
-  const checkedConsumerDelegations = false //TODO disponibilità fruizione
+  const [checkedConsumerDelegations, setCheckedConsumerDelegations] = React.useState(
+    isAvailableConsumerDelegations
+  )
+
+  React.useEffect(() => {
+    if (!isOpen) {
+      if (checkedProducerDelegations != isAvailableProducerDelegations) {
+        setCheckedProducerDelegations(isAvailableProducerDelegations)
+      }
+
+      if (checkedConsumerDelegations != isAvailableConsumerDelegations) {
+        setCheckedConsumerDelegations(isAvailableConsumerDelegations)
+      }
+    }
+  }, [
+    isAvailableProducerDelegations,
+    isAvailableConsumerDelegations,
+    isOpen,
+    checkedProducerDelegations,
+    checkedConsumerDelegations,
+  ])
 
   function handleClick() {
```

Comment:
Let's make this more specific

```suggestion
  function handleSubmit() {
```